js/src/Ice/TimerUtil.js

```diff
@@ -4,15 +4,246 @@
 
 const Ice = require("../Ice/ModuleRegistry").Ice;
 
-class Timer
+let Timer = {};
+
+if (typeof process != 'undefined')
 {
+    Timer = class
+    {
+    }
+
+    Timer.setTimeout = setTimeout;
+    Timer.clearTimeout = clearTimeout;
+    Timer.setInterval = setInterval;
+    Timer.clearInterval = clearInterval;
+    Timer.setImmediate = setImmediate;
+    Ice.Timer = Timer;
 }
+else
+{
+    function isIE()
+    {
+        return (navigator.userAgent.indexOf("MSIE") !== -1 || navigator.userAgent.match(/Trident.*rv:11\./));
+    }
+
+    function isEdge()
+    {
+        return (/Edge/).test(navigator.userAgent);
+    }
+
+    function isWorker()
+    {
+        return typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope;
+    }
+
+    const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 9007199254740991;
+    let _nextId = 0;
+    const nextId = function()
+    {
+        if(_nextId == MAX_SAFE_INTEGER)
+        {
+            _nextId = 0;
+        }
+        return _nextId++;
+    };
+    const _timers = new Map();
+
+    //
+    // Create a timer object that uses the default browser methods.
+    //
+    function createTimerObject()
+    {
+        const Timer = class
+        {
+            static setTimeout(cb, ms)
+            {
+                return setTimeout(cb, ms);
+            }
+
+            static clearTimeout(id)
+            {
+                return clearTimeout(id);
+            }
+
+            static setInterval(cb, ms)
+            {
+                return setInterval(cb, ms);
+            }
+
+            static clearInterval(id)
+            {
+                return clearInterval(id);
+            }
+        };
+
+        //
+        // For Browsers that support setImmediate prefer that,
+        // otherwise implement it using MessageChannel
+        //
+        if(isEdge() || isIE())
```

Comment:
Does edge have a JS feature that other chromium browsers do not have? Or is this about old Edge, and if so should we remove it?

Reply:
Seems like a check we have for old edge, we can probably remove that too

Reply:
So the new version uses `Edg/` token instead of `Edge/` and will not match this check, see https://docs.microsoft.com/en-us/microsoft-edge/web-platform/user-agent-guidance